tests/ui/integer_arithmetic.rs

```diff
@@ -17,6 +17,10 @@ fn main() {
     i / 2; // no error, this is part of the expression in the preceding line
     i - 2 + 2 - i;
     -i;
+    i / 0;
+    i / -1;
+    i % 0;
+    i % -1;
```

Comment:
Could you also add tests with a non-literal divisor to be sure that the lint is triggered in that case?

Reply:
done.